pandas/core/reshape/pivot.py

```diff
@@ -455,17 +455,22 @@ def crosstab(index, columns, values=None, rownames=None, colnames=None,
 
     from pandas import DataFrame
     df = DataFrame(data, index=common_idx)
+    original_columns_set = set(df.columns.values)
     if values is None:
         df['__dummy__'] = 0
         kwargs = {'aggfunc': len, 'fill_value': 0}
     else:
         df['__dummy__'] = values
         kwargs = {'aggfunc': aggfunc}
+    added_column = list(set(df.columns.values) - original_columns_set)[0]
 
-    table = df.pivot_table('__dummy__', index=rownames, columns=colnames,
+    table = df.pivot_table(['__dummy__'], index=rownames, columns=colnames,
                            margins=margins, margins_name=margins_name,
                            dropna=dropna, **kwargs)
 
+    if not table.empty:
+        table = table[added_column]
```

Comment:
so you don't need all of the computation of the added_column to be above, just here.

also, just use the Index operations themselves.

``df.columns.difference(common_columns)``

Reply:
Noted. I'll look into this operation